test/unit/org/apache/cassandra/cql3/validation/operations/InsertTest.java

```diff
@@ -247,4 +252,42 @@ public void testCKInsertWithValueOver64K() throws Throwable
         assertInvalidThrow(InvalidRequestException.class,
                            "INSERT INTO %s (a, b) VALUES ('foo', ?)", new String(TOO_BIG.array()));
     }
+
+    @Test
+    public void testInsertingMapDataWithParameterizedQueriesIsKeyOrderIndependent() throws Throwable
```

Comment:
This test is mostly identical to the one on `SelectTest`, we could combine them and put them in `CollectionsTest.java`.

Reply:
I actually originally split the two tests out specifically so we could test inserting with parameterized queries and selecting with parameterized queries specifically (note that each test tests parameterizing a different query). Without this, you could theoretically break one side without being able to test it on the other side.
I'm happy to move it to `CollectionsTest.java` if you feel it's a more appropriate place for them to live, but I think it's important to make sure there are two different, distinct tests so we cover both.

Reply:
Moving all of these tests, plus the new ones for sets, into CollectionTests.